drizzlepac/haputils/svm_quality_analysis.py

```diff
@@ -516,6 +516,304 @@ def compare_ra_dec_crossmatches(hap_obj, json_timestamp=None, json_time_since_ep
+                else:
+                    filt_names = "{} - {}".format(filtername_comp, filtername_ref)
+                    log.warning("{} interfilter cross match test could not be performed.".format(filt_names))
+
+    # Housekeeping. Delete the *_point-cat-fxm.ecsv files created for cross-matching, and the
+    # filtobj_dict dictionary
+    for temp_cat_filename in temp_cat_file_list:
+        log.info("removing temp catalog file {}".format(temp_cat_filename))
+        os.remove(temp_cat_filename)
+    del filtobj_dict
+
+
+# ------------------------------------------------------------------------------------------------------------
+
+
+def transform_coords(filtobj_subdict, xmatch_ref_imgname, log_level=logutil.logging.NOTSET):
+    """Transform comparision image frame of reference x, y coords to RA and dec, then back to x, y coords in
+    the cross-match reference image's frame of reference
+
+    Parameters
+    ----------
+    filtobj_subdict : dict
+        dictionary containing a drizzlepac.haputils.Product.FilterProduct and corresponding source catalog
+        generated earlier in the run by **find_hap_point_sources**.
+
+    xmatch_ref_imgname : str
+        name of the reference image whose WCS info will be used to convert RA and dec values into x, y coords
+        in a common frame
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the
+        .log file. Default value is 'NOTSET'.
+
+    Returns
+    -------
+    filtobj_dict : dict
+        the input **filtobj_dict** dictionary with the 'sources' table updated to include freshly computed ra,
+        dec, x_centroid_ref, and y_centroid_ref columns
+    """
+    # Initiate logging!
+    log.setLevel(log_level)
+
+    # 1: stack up xcentroid and ycentorid columns from sources table
+    xy_centroid_values = np.stack((filtobj_subdict['sources']['xcentroid'],
+                                   filtobj_subdict['sources']['ycentroid']), axis=1)
+
+    # 2: perform coordinate transforms.
+    origin = 0
+    fits_exten = "[1]"
+    imgname = filtobj_subdict['filt_obj'].drizzle_filename
+
+    # 2a: perform xcentroid, ycentroid -> ra, dec transform
+    ra_dec_values = hla_flag_filter.xytord(xy_centroid_values, imgname, fits_exten, origin=origin)
+
+    # 2b: perform ra, dec -> x_centroid_ref, y_centroid_ref transform
+    xy_centroid_ref_values = hla_flag_filter.rdtoxy(ra_dec_values, xmatch_ref_imgname, fits_exten,
+                                                    origin=origin)
+
+    # 3: add new columns to filtobj_subdict['sources'] table
+    title_data_dict = collections.OrderedDict()
+    title_data_dict["ra"] = ra_dec_values[:, 0]
+    title_data_dict["dec"] = ra_dec_values[:, 1]
+    title_data_dict["xcentroid_ref"] = xy_centroid_ref_values[:, 0]
+    title_data_dict["ycentroid_ref"] = xy_centroid_ref_values[:, 1]
```

Comment:
It may be useful to add the xmatch_ref_imgname plate scale to this dictionary so it can be passed along to the JSON file later for interpretation of the pixel differences.

Reply:
issue resolved by commit [f0bb5ad](https://github.com/spacetelescope/drizzlepac/pull/719/commits/f0bb5ad7c371aecf7bd615393618889fe2c1aa90)